docs/releases/upgrading/upgrade-v12.0.md

```diff
@@ -0,0 +1,215 @@
+```
+
+#### Option 2
+
+This approach will create a new virtual environment and leaves the v11 as it is.
+If you are using a docker image on your production instance this will be the
+option you chose.
+
+##### Step 1
+- create a new virtual environment
+- activate your new virtual environment
+- install `invenio-cli` by `pip install invenio-cli`
+
+##### Step 2
+Update the file `<my-site>/Pipfile`.
+
+```diff
+[packages]
+---invenio-app-rdm = {extras = [...], version = "~=11.0.0"}
++++invenio-app-rdm = {extras = [...], version = "~=12.0.0"}
+```
+
+Due to a dependency upgrade, update the following line.
+```diff
+---my-site = {editable="True", path="./site"}
++++my-site = {editable=true, path="./site"}
+```
+
+##### Step 3
+Update the `Pipfile.lock` file
+
+```bash
+invenio-cli packages lock
+```
+
+##### Step 4
+install InvenioRDM v12
+
+```bash
+invenio-cli install
+```
+
+
+### Database migration
+
+Execute the database migration:
+
+```bash
+# Execute the database migration
+invenio alembic upgrade
+```
+
+### Declare usage statistics processing queues
+
+```shell
+invenio queues declare
+```
+
+### Update indices mappings
+
+```shell
+invenio index update communities-communities-v1.0.0
+invenio index update rdmrecords-drafts-draft-v5.0.0
+invenio index update rdmrecords-records-record-v5.0.0
+```
+
+CAVEAT: this is not working because of a permission problem in
+[invenio-search](https://github.com/inveniosoftware/invenio-search/blob/d8b23ecf48f63d8d313f90fd4618a480e15fbd7b/invenio_search/ext.py#L448).
+The only way to solve it is to destroy the index and reinit and rebuild it from
+scratch. Yes the statistics will be only stored in the index but the statistic
+indices are created by templates and therefore not affected by `invenio index
+destroy --yes-i-know`. The statistic indices will remain in the list of indices
+and are totally functional also after a reinit.
+
+
+### Data migration
+
+Execute the data migration
+
+```bash
+pipenv run invenio shell $(find $(pipenv --venv)/lib/*/site-packages/invenio_app_rdm -name migrate_11_0_to_12_0.py)
+```
+
+### Reindex
+
+```bash
+invenio index destroy --yes-i-know
+invenio index init
+invenio rdm rebuild-all-indices
+```
+
+### New roles
+
+```bash
+invenio roles create administration-moderation
+invenio roles create administration
+
+invenio access allow administration-moderation role administration-moderation
+invenio access allow administration-access role administration
+invenio access allow superuser-access role administration
+```
+
+### New configuration variables
+
+```bash
+COMMUNITIES_IDENTITIES_CACHE_REDIS_URL = "URI_TO_REDIS"
+USERS_RESOURCES_ADMINISTRATION_ENABLED = True
+THEME_SITENAME = "Project name for header and UI"
+```
+
+## Big Changes
+
+- remove: dependency of flask-babelex
+- add: concept doi (aka parent doi)
+- add: statistics. (NOTE: statistic is stored in opensearch/libresearch)
+- add: administration panel
+- add: set quota
+- add: branded communities
```

Comment:
add: share options for drafts and records

Reply:
done